v2/tools/generator/internal/functions/get_extended_resources_function.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astbuilder"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/dave/dst"
+	"sort"
+)
+
+const (
+	ExtendedResourcesFunctionName = "GetExtendedResources"
+)
+
+type GetExtendedResourcesFunction struct {
+	idFactory astmodel.IdentifierFactory
+	resources []astmodel.TypeName
+}
+
+// Ensure GetExtendedResources properly implements Function
+var _ astmodel.Function = &GetExtendedResourcesFunction{}
+
+// NewGetExtendedResourcesFunction creates a new GetExtendedResources
+func NewGetExtendedResourcesFunction(idFactory astmodel.IdentifierFactory, resources []astmodel.TypeName) *GetExtendedResourcesFunction {
+
+	result := &GetExtendedResourcesFunction{
+		idFactory: idFactory,
+		resources: sortResources(resources),
+	}
+	return result
+}
+
+//getPackageRefs iterates through the resources and returns package references
```

Comment:
minor, style: spaces before comments here too (and below -- won't repeat though).